Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -5845,25 +5870,67 @@ webkit_web_view_get_default_content_security_policy(WebKitWebView* webView)
     return webView->priv->defaultContentSecurityPolicy.data();
 }
 
+void webkitWebViewSetIsImmersiveModeEnabled(WebKitWebView* webView, bool isImmersiveModeEnabled)
+{
+#if ENABLE(WEBXR)
+    if (webView->priv->isImmersiveModeEnabled == isImmersiveModeEnabled)
+        return;
+
+    webView->priv->isImmersiveModeEnabled = isImmersiveModeEnabled;
+    g_object_notify_by_pspec(G_OBJECT(webView), sObjProperties[PROP_IS_IMMERSIVE_MODE_ENABLED]);
+#endif
```

Comment:
Since we have guards for every cal tot this, it's better to define it only for WEBXR too and we avoid warnings for unused parameters when it's disabled.